mlflow/azureml/__init__.py

```diff
@@ -206,8 +207,163 @@ def build_image(model_uri, workspace, image_name=None, model_name=None,
+    # upon method invocation.
+    # pylint: disable=import-error
+    from azureml.core.model import Model as AzureModel, InferenceConfig
+    from azureml.core import Environment as AzureEnvironment
+    from azureml.core import VERSION as AZUREML_VERSION
+
+    absolute_model_path = _download_artifact_from_uri(model_uri)
+
+    model_pyfunc_conf, run_id = _load_pyfunc_conf(model_path=absolute_model_path)
+    model_python_version = model_pyfunc_conf.get(pyfunc.PY_VERSION, None)
+    if model_python_version is not None and\
+            StrictVersion(model_python_version) < StrictVersion("3.0.0"):
+        raise MlflowException(
+                message=("Azure ML can only deploy models trained in Python 3 and above. See"
+                         " the following MLflow GitHub issue for a thorough explanation of this"
+                         " limitation and a workaround to enable support for deploying models"
+                         " trained in Python 2: https://github.com/mlflow/mlflow/issues/668"),
+                error_code=INVALID_PARAMETER_VALUE)
+
+    tags = _build_tags(model_uri=model_uri, model_python_version=model_python_version)
+
+    if service_name is None:
+        service_name = _get_mlflow_azure_name_from_run(run_id)
+    if model_name is None:
+        model_name = _get_mlflow_azure_name_from_run(run_id)
+
+    with TempDir(chdr=True) as tmp:
+        model_directory_path = tmp.path("model")
+        tmp_model_path = os.path.join(
+            model_directory_path,
+            _copy_file_or_tree(src=absolute_model_path, dst=model_directory_path))
+
+
+        registered_model = AzureModel.register(workspace=workspace, model_path=tmp_model_path,
+                                               model_name=model_name, tags=tags)
+
+        _logger.info("Registered an Azure Model with name: `%s` and version: `%s`",
+                     registered_model.name, registered_model.version)
+
+        # Create an execution script (entry point) for the image's model server. Azure ML requires
+        # the container's execution script to be located in the current working directory during
+        # image creation, so we create the execution script as a temporary file in the current
+        # working directory.
+        execution_script_path = tmp.path("execution_script.py")
+        _create_execution_script(output_path=execution_script_path, azure_model=registered_model)
+
+        environment = None
+        if pyfunc.ENV in model_pyfunc_conf:
+            environment = AzureEnvironment.from_conda_specification(_get_mlflow_azure_name_from_run(run_id), os.path.join(tmp_model_path, model_pyfunc_conf[pyfunc.ENV]))
+        else:
+            raise MlflowException("Cannot deploy pyfunc model without an environment to Azure ML.")
+
+        if mlflow_home is not None:
+            wheel = _create_mlflow_wheel(mlflow_home, tmp.path("dist"))
```

Comment:
Nit: Can we emit a log message indicating that we are about to build an MLflow wheel before beginning the build process?